scipy/optimize/_basinhopping.py

```diff
@@ -3,12 +3,18 @@
 """
 import numpy as np
 import math
+import inspect
 import scipy.optimize
 from scipy._lib._util import check_random_state
 
 __all__ = ['basinhopping']
 
 
+_params = (inspect.Parameter('res_new', kind=inspect.Parameter.KEYWORD_ONLY),
+           inspect.Parameter('res_old', kind=inspect.Parameter.KEYWORD_ONLY))
+_new_accept_test_signature = inspect.Signature(parameters=_params)
```

Comment:
Currently, the new signature is exactly two keyword-only arguments: `res_new, res_old`. This seemed to be a natural generalization of the existing `f_new, x_new, f_old, x_old`. If the user's `accept_test` has exactly the new signature, it will be passed two `OptimizeResult`s instead of `f_new, x_new, f_old, x_old`.